src/app/content/highlights/reducer.ts

```diff
@@ -35,8 +38,22 @@ export const initialState: State = {
 const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
   switch (action.type) {
     case getType(locationChange): {
-      return {...initialState, myHighlightsOpen: false,
-        summary: {...state.summary},
+      const currentPageId = state.currentPage.pageId;
+      const actionPageId = action.payload.location.state && action.payload.location.state.pageUid;
+      return {
+        currentPage: {
+          ...initialState.currentPage,
+          highlights: currentPageId && actionPageId === currentPageId
+            ? state.currentPage.highlights
+            : initialState.currentPage.highlights,
+          pageId: currentPageId && actionPageId === currentPageId
+            ? state.currentPage.pageId
+            : initialState.currentPage.pageId,
```

Comment:
these two use the same condition, so you could push it up and do 
```
return {
        currentPage: currentPageId && actionPageId === currentPageId
            ? {...state.currentPage, hasUnsavedHighlight: false}
            : initialState.currentPage
```

Reply:
Thanks, I've pushed a change